modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,282 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-kuryr-sdn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-kuryr-cleanup_{context}"]
+= Cleaning up resources after migration
+
+After migration from the Kuryr network plugin to the OVN-Kubernetes network plugin completes, clean up the resources that Kuryr created previously. 
+
+.Prerequisites
+
+* You installed the {product-title} CLI, `oc`.
+* You installed a Python interpreter.
+* You installed the `openstacksdk` Python package.
+* You installed the `openstack` CLI.
+* You have access to the underlying {rh-openstack} cloud.
+* You can access the cluster as a user with the `cluster-admin` role.
+
+[NOTE]
+====
+At the time of writing when OCP16 is used, it includes ancient version of
+`openstacksdk` package, that does not support tags for Octavia objects. It is
+recommended to use either workstation with modern operating system with newer
+version of `openstacksdk` package or `virtualenv` and install appropriate
+version from link:https://pypi.org[pypi].
+====
+
+.Setup virtualenv
+
+If you decide to use virtualenv, here is a quick guidance how to proceed.
+At the time of writing, minimal versions of the following OpenStack components
+has been used to fulfill cleaning procedure:
+
+- openstacksdk 0.54.0
+- python-openstackclient 5.5.0
+- python-octaviaclient 2.3.0
+
+Newer versions should also work, although there is a tiny chance for
+incompatibilities. If you have issues, either downgrade to version mentioned
+above or consult individual component documentation and/or changelog for
+details.
+
+To set up cleanup environment, create virtualenv, update pip and install needed
+components:
+
+[source,terminal]
+----
+$ python3 -m venv /tmp/venv
+$ source /tmp/venv/bin/activate
+(venv) $ pip install pip --upgrade
+(venv) $ pip install openstacksdk==0.54.0 python-openstackclient==5.5.0 python-octaviaclient==2.3.0
+----
+
+[NOTE]
+====
+In the command prompt there will be visible indicator, that virtualenv is in
+use - here it will be additional `(venv)` string, as in the example above venv
+is a name of the directory and the virtual env at the same time.
+====
+
+Stay in the created environment during following procedure to be sure the
+right python components are used.
+
+.Procedure
```

Comment:
```suggestion
.Procedure

. Create a clean-up Python virtual environment:

.. Create a temporary directory for your environment. For example, with `venv`:
+
[source,terminal]
----
$ python3 -m venv /tmp/venv
----
+ 
`venv` is used in all clean up examples.

.. Enter the virtual environment. For example:
+
[source,terminal]
----
$ source /tmp/venv/bin/activate
----

.. Upgrade `pip` in the virtual environment by running the following command:
+
[source,terminal]
----
(venv) $ pip install pip --upgrade
---- 

.. Install the required Python packages by running the following command:
+
[source,terminal]
----
(venv) $ pip install openstacksdk==0.54.0 python-openstackclient==5.5.0 python-octaviaclient==2.3.0
----
```

@gryf @imatza-rh Thoughts on this? I'm wondering whether it's necessary to specify venv or just direct the reader to use whichever virtual environment tool they like. Shouldn't matter as long as the packages get installed, right?

It also seems safe to drop the packages installed in the venv instructions from the prerequisites.

Reply:
+1

Right, the goal is to have an environment with the appropriate packages, and it doesn't matter if it's venv (or which kind of virtual environment tool).

On the other side, we should keep the note, so the user knows he can keep using his env if it supports tags for Octavia objects.
```
[NOTE]
====
At the time of writing when OCP16 is used, it includes ancient version of
`openstacksdk` package, that does not support tags for Octavia objects. It is
recommended to use either workstation with modern operating system with newer
version of `openstacksdk` package or `virtualenv` and install appropriate
version from link:https://pypi.org[pypi].
====
```

Reply:
I agree with @imatza-rh here. Let's leave the note which explains why we are using virtual env for the procedure, and what are the other options.

Reply:
I leaved the note about reasons and rearranged beginning of the section as you suggested, @maxwelldb